### PR TITLE
MAGECLOUD-4813: Stabilizing EOL Validation unit tests.

### PR DESCRIPTION
**Pull request by sikharam-adb**
### Description

Stabilizing unit test results for service EOL validation.

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-4813

### Manual testing scenarios

https://jira.corp.magento.com/browse/MAGECLOUD-141

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Test/Unit/Service/EolValidatorTest.php [hunk 1]
```diff
@@ -69,6 +69,8 @@ protected function setUp()
      */
     public function testCompatibleVersion()
     {
+        Carbon::setTestNow(Carbon::create(2019, 12, 2));
```
Comment (bbatsche) [inline src/Test/Unit/Service/EolValidatorTest.php]: Move all of these to the `setUp()` method

File: src/Test/Unit/Service/EolValidatorTest.php [hunk 2]
```diff
@@ -84,13 +86,17 @@ public function testCompatibleVersion()
         $serviceVersion = '6.5';
 
         $this->assertEquals([], $this->validator->validateService($serviceName, $serviceVersion));
+
+        Carbon::setTestNow();
```
Comment (bbatsche) [inline src/Test/Unit/Service/EolValidatorTest.php]: Create a `tearDown()` method and put these in there.

**Review by bbatsche — CHANGES_REQUESTED** [review]

**Review by billygilbert — CHANGES_REQUESTED** [review]
Agreed with @bbatsche's comments.

File: src/Test/Unit/Service/EolValidatorTest.php [hunk 3]
```diff
@@ -169,6 +179,8 @@ public function testValidateServiceEol()
      */
     public function testValidateNoticeMessage()
     {
+        Carbon::setTestNow(Carbon::create(2019, 11, 1));
```
Comment (billygilbert) [inline src/Test/Unit/Service/EolValidatorTest.php:182]: Does this need to be here or could you just change the test date to be later?

Comment (sikharam-adb) [reply src/Test/Unit/Service/EolValidatorTest.php:182]: This is the only test where a different 'test date' is required to produce the desired result. That, or I could remove this line and modify the EOL for the service in the _file/eol_2.yaml file. I figured setting a different EOL for the service in the config file for the purposes of the test may cause confusion, but either way works.

Comment (billygilbert) [reply src/Test/Unit/Service/EolValidatorTest.php:182]: kk, then makes sense to leave as is.

**Review by billygilbert — APPROVED** [review]

**Review by bbatsche — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Comment by shiftedreality** [comment]
No QA is needed

**Pull request merged**